lib/src/main/java/com/smileidentity/util/FileUtils.kt

```diff
@@ -0,0 +1,496 @@
+const val UPLOAD_REQUEST_FILE = "info.json"
+
+// Enum defining the types of files managed within the job processing system.
+// This categorization helps in filtering and processing files based on their content or purpose.
+
+/**
+ * Represents the types of files associated with a job. Each type has specific roles
+ * within the job processing workflow:
+ * - SELFIE: Refers to selfie capture file captured that is pertinent to the job
+ * - LIVENESS: Refers to liveness images captured file captured that is pertinent to the job
+ * - DOCUMENT:Refers to document capture files captured that is pertinent to the job
+ */
+enum class FileType(val fileType: String) {
+    SELFIE("si_selfie_"),
+    LIVENESS("si_liveness_"),
+    DOCUMENT_FRONT("si_document_front_"),
+    DOCUMENT_BACK("si_document_back_"),
+    ;
+
+    override fun toString(): String {
+        return fileType
+    }
+}
+
+/**
+ * Performs cleanup of job files based on specified criteria, allowing selective deletion of files
+ * associated with completed and/or pending jobs. This function provides granular control over the
+ * cleanup process, enabling targeted deletion operations based on the job's completion status and
+ * optionally, specific job IDs.
+ *
+ * @param deleteSubmittedJobs When set to true, the function targets files associated with completed
+ * jobs for deletion. This enables clearing out space or managing files that are no longer needed
+ * after job completion. Defaults to false to prevent accidental deletion of completed jobs.
+ *
+ * @param deleteUnsubmittedJobs When set to true, the function targets files associated with unsubmitted
+ * jobs for deletion. Useful for resetting or clearing jobs that have not been completed or are
+ * no longer needed. Defaults to false to protect ongoing or queued jobs from unintended deletion.
+ *
+ * @param jobIds An optional list of specific job IDs to delete. If provided, the cleanup process
+ * is limited to these IDs, within the context of the completion status flags set. If null, the
+ * function applies to all jobs within the specified completion status categories, allowing for
+ * bulk cleanup operations.
+ *
+ * This function directly manipulates the file system by removing files and directories associated
+ * with the targeted jobs. It's designed to facilitate efficient storage management and job lifecycle
+ * maintenance within the system, ensuring that resources are allocated and used effectively.
+ */
+internal fun cleanupJobs(
+    deleteSubmittedJobs: Boolean = false,
+    deleteUnsubmittedJobs: Boolean = false,
+    jobIds: List<String>? = null,
+    // Default to the base save path used by createSmileTempFile
+    savePath: String = SmileID.fileSavePath,
+) {
+    val pathsToClean = mutableListOf<String>()
+    if (deleteSubmittedJobs) pathsToClean.add("$savePath/$SUBMITTED_PATH")
+    if (deleteUnsubmittedJobs) pathsToClean.add("$savePath/$UNSUBMITTED_PATH")
+
+    if (jobIds.isNullOrEmpty()) {
```

Comment:
If jobIds is empty, we shouldn't delete everything! We only nuke everything if jobIds is null

```suggestion
    if (jobIds == null) {
```